jobs/build-snaps/build-release-cdk-addons.groovy

```diff
@@ -88,76 +86,59 @@ pipeline {
+                 */
+                sh """#!/usr/bin/env bash
+                    . \${WORKSPACE}/cilib.sh
+
+                    ci_lxc_launch ubuntu:20.04 ${lxc_name}
+                    until sudo lxc shell ${lxc_name} -- bash -c "snap install snapcraft --classic"; do
+                        echo 'retrying snapcraft install in 3s...'
+                        sleep 3
+                    done
+                    sudo lxc shell ${lxc_name} -- bash -c "apt-get install containerd -y"
+
+                    # we'll upload snaps from within the container; put creds in place
+                    sudo lxc file push /var/lib/jenkins/snapcraft-creds ${lxc_name}/snapcraft-creds
+                """
+            }
+        }
+        stage('Build Snaps'){
             steps {
                 echo "Setting K8s version: ${kube_version} and K8s ersion: ${kube_ersion}"
                 sh """
                     cd cdk-addons
-                    # prep ./build/templates and track upstream images
-                    UPSTREAM_KEY=${kube_version}-upstream:
-                    UPSTREAM_LINE=\$(make KUBE_VERSION=${kube_version} prep upstream-images 2>/dev/null | grep ^\${UPSTREAM_KEY})
+                    make KUBE_VERSION=${kube_version} prep 2>/dev/null
 
-                    # build snap
                     for arch in ${env.ADDONS_ARCHES}
                     do
-                        echo "Building cdk-addons snap for arch \${arch}."
+                        echo "Prepping cdk-addons (\${arch}) snap source."
                         wget -O build/kubectl https://storage.googleapis.com/kubernetes-release/release/${kube_version}/bin/linux/\${arch}/kubectl
                         chmod +x build/kubectl
                         sed 's/KUBE_VERSION/${kube_ersion}/g' cdk-addons.yaml > build/snapcraft.yaml
                         if [ "\${arch}" = "ppc64le" ]
                         then
-                          sed -i "s/KUBE_ARCH/ppc64el/g" build/snapcraft.yaml
-                          cd build
-                          SNAPCRAFT_BUILD_ENVIRONMENT=host snapcraft clean || exit 1
-                          SNAPCRAFT_BUILD_ENVIRONMENT=host snapcraft --target-arch=ppc64el || exit 1
-                          mv *.snap ..
-                          cd ..
-                        else
-                          sed -i "s/KUBE_ARCH/\${arch}/g" build/snapcraft.yaml
-                          cd build
-                          SNAPCRAFT_BUILD_ENVIRONMENT=host snapcraft clean || exit 1
-                          SNAPCRAFT_BUILD_ENVIRONMENT=host snapcraft --target-arch=\${arch} || exit 1
-                          mv *.snap ..
-                          cd ..
+                          arch="ppc64el"
                         fi
-                    done
-                    cd ..
+                        sed -i "s/KUBE_ARCH/\${arch}/g" build/snapcraft.yaml
 
-                    echo "Updating bundle with upstream images."
```

Comment:
The purpose of this job is to pull CK images from our staging area, then tag/push them to our production area. There is no need to manipulate the bundle's `container-images.txt` file in this job because it already (rightfully) happens in the [sync-oci-images](https://github.com/charmed-kubernetes/jenkins/blob/98a89b0166a9a94211a1c8804c6e65192f667c53/jobs/sync-oci-images/sync-oci-images.groovy#L75-L108) job.